### PR TITLE
🎁 IOTSTOOL2039 - Foundation support for filters

### PR DESCRIPTION
**Pull request by alexl0gan**

### CODE REVIEW

File: generator/common/utilities.ts [hunk 1]
```diff
@@ -94,6 +94,23 @@ const safeAddToList = (list: Array<Container>, item: Container): void => {
     }
 };
 
+const ensureArray = obj => {
```
Comment (mark-edgeworth) [inline generator/common/utilities.ts]: Remove, rename or recode

File: generator/common/utilities.ts [hunk 2]
```diff
@@ -94,6 +94,23 @@ const safeAddToList = (list: Array<Container>, item: Container): void => {
...
+    if (isEmpty(obj)) {
+        return [];
+    } else {
+        return obj;
+    }
+};
+
+const isEmpty = obj => {
```
Comment (mark-edgeworth) [inline generator/common/utilities.ts]: Get rid

File: src/common/pagination.ts [hunk 3]
```diff
@@ -217,7 +217,15 @@ export class Paginator<T, U extends ListOptions> {
 
     private browseAndConcatenateAllPages(): Promise<Array<T>> {
         if (this.hasNewPage()) {
-            return this.nextPage().then( page => this.browseAndConcatenateAllPages().then( data => page.data.length === 0 ? data : page.data.concat(data)));
+            return this.nextPage()
```
Comment (mark-edgeworth) [inline src/common/pagination.ts]: Raise bug on this

File: test/integration/method/method.ts [hunk 4]
```diff
@@ -115,6 +115,9 @@ export class Method {
             } else if (parameter.type === "file") {
                 // if file then return a read stream
                 return fs.createReadStream(testRunnerParameters[foundParam]);
+            } else if (parameter.type.indexOf("Filter") > -1) {
```
Comment (mark-edgeworth) [inline test/integration/method/method.ts]: Use endsWith

File: generator/common/utilities.ts (hunk 1 above)
Comment (alexl0gan) [reply generator/common/utilities.ts]: This wasn't used so removed

File: generator/common/utilities.ts (hunk 2 above)
Comment (alexl0gan) [reply generator/common/utilities.ts]: Renamed to isEmptyFilter

File: src/common/pagination.ts (hunk 3 above)
Comment (alexl0gan) [reply src/common/pagination.ts]: https://jira.arm.com/browse/IOTSTOOL-2296

**Pull request merged**